src/lib/transaction_snark/test/verify-simple-test/transaction_snark_tests_verify_simple_test.ml

```diff
@@ -0,0 +1,22 @@
+open Mina_base
```

Comment:
This test-case reproduces the failure seen [here](https://buildkite.com/o-1-labs-2/mina/builds/25913#01848366-40ce-4959-b27d-7538a9625df5) in integration tests on CI, in a far more compact, debuggable way.

This could be reduced further, and placed inside the pickles module.

The issue we find is that
* if the inferred wrap domain is incorrect,
* and the verification key is side-loaded,
* and that side-loaded key was deserialised before being used,

then the `wrap_vk` is not serialised, but generated using the number of previous proofs.

The current workaround relies on getting the correct domain from the `wrap_vk` when it doesn't match the inferred one for the number of previous proofs. This doesn't work in this situation; we will have to explicitly pass around the wrap domain as part of the vk instead.